cdap-ui/app/cdap/components/EntityListView/index.js

```diff
@@ -135,23 +135,22 @@ class EntityListView extends Component {
      );
     }
 
-    let query = nextProps.location.query || {};
-    let {filter = defaultFilter, q , page = 1, sort, order } = query;
-    let sortObj;
-    if (typeof q === 'undefined'){
-      let matchedSort = this.sortOptions.find(sortOption => sortOption.sort === sort && sortOption.order === order);
-      sortObj = matchedSort ? matchedSort : this.sortOptions[0];
-    } else {
-      sortObj = this.sortOptions[0];
+    let queryObject = this.getQueryObject(nextProps.location.query);
+    if (
+      queryObject.filter !== this.state.filter &&
+      queryObject.sort.fullSort !== this.state.sortObj.fullSort &&
+      queryObject.query !== this.state.query &&
+      queryObject.page !== this.state.currentPage
+    ) {
+      this.updateData(queryObject.query, queryObject.filter, queryObject.sort, nextProps.params.namespace, queryObject.page);
```

Comment:
So this is part of react's life cycle methods. When the user goes to `/` url the `componentWillReceiveProps` function won't be invoked at all. But if they go to `/cdap/ns/default` this will be invoked as its being mounted and it receives new props based on the url. So thats the reason I am checking here if the props have changed. 

Ideally we shouldn't be making this check and have one flow to update the UI and the URL. Will file a JIRA to fix this in 4.1.